source/server/config_validation/server.cc

```diff
@@ -42,9 +42,10 @@ ValidationInstance::ValidationInstance(
     const Network::Address::InstanceConstSharedPtr& local_address, Stats::IsolatedStoreImpl& store,
     Thread::BasicLockable& access_log_lock, ComponentFactory& component_factory,
     Thread::ThreadFactory& thread_factory, Filesystem::Instance& file_system)
-    : options_(options), validation_context_(options_.allowUnknownStaticFields(),
-                                             !options.rejectUnknownDynamicFields(),
-                                             !options.ignoreUnknownDynamicFields()),
+    : ProdListenerComponentFactory(*dynamic_cast<Server::Instance*>(this)), options_(options),
```

Comment:
prefer `static_cast` as it's cheaper.  It seems like the issue without a cast is that it's ambigious to the compiler which ctor you want to invoke for this.

Reply:
Yes, make sense.

For a couple of reasons, we decided to take an alternative approach here to setup a local tcp_listener_config_provider_manager_ instance in class ValidationInstance to be used in  the function call: 
ProdListenerComponentFactory::createListenerFilterFactoryListImpl(
        filters, context, *tcp_listener_config_provider_manager_);
With that, call ValidationInstancethis doesn't have to be derived from ProdListenerComponentFactory, thus this problem can be avoided.